test/airportTestSuite.spec.js

```diff
@@ -0,0 +1,289 @@
+airport.planesStanding = []
+airport.airportCapacity = 10;
+result = undefined;
+actual = undefined;
+
+//------------------------------------------------------------ 6
+console.log(`
+Test 6
+==============================================
+Planes do not land if the airport is at capacity:
+    Ensure airport has a capacity property and this limits planes landing
+`);
+
+// Arrange
+airport.planesStanding = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+expected = airport.airportCapacity;
+
+// Act
+airport.landPlane(plane);
+actual = airport.planesStanding.length;
+
+// Assert
+result = assertEquals(actual, expected);
+
+// Report
+console.log(`Test 6: planes can't land when the airport is full: ${result}`);
+console.log(`Test 6: ${result ? `PASS` : `FAIL`}`);
+console.log(``);
+
+// Clean Up
+airport.planesStanding = [];
+result = undefined;
+actual = undefined;
+
+//------------------------------------------------------------7
+console.log(`
+Test 7
+==============================================
+Planes can take off when instructed:
+    Ensure planes can leave the airport
+`);
+
+// Arrange
+airport.landPlane(plane);
+expected = 0;
+
+// Act
+airport.orderTakeOff(plane);
+actual = airport.planesStanding.length;
+
+// Assert
+result = assertEquals(actual, expected);
+
+// Report
+console.log(`Test 7: planes leave the airport when instructed: ${result}`);
+console.log(`Test 7: ${result ? `PASS` : `FAIL`}`);
+console.log(``);
+
+// Clean Up
+airport.planesStanding = [];
+result = undefined;
+actual = undefined;
+
+//------------------------------------------------------------8
+console.log(`
+Test 8
+==============================================
+The right plane leaves the airport when instructed:
+   Ensure the right plane leaves the airport
+`);
+
+// Arrange
+plane2 = new Plane(`plane2`);
+airport.landPlane(plane2);
+airport.landPlane(plane);
+expected = 0;
+
+// Act
+airport.orderTakeOff(plane);
+
+// Assert
+result = assertEquals(airport.planesStanding.includes(plane), false);
+
+// Report
+console.log(`Test 8: the right plane leaves the airport: ${result}`);
+console.log(`Test 8: ${result ? `PASS` : `FAIL`}`);
+console.log(``);
+
+// Clean Up
+airport.planesStanding = [];
+result = undefined;
+actual = undefined;
+
+//------------------------------------------------------------9
+console.log(`
+Test 9
+==============================================
+Planes that aren't at the airport can't leave:
+   Ensure that only planes at the airport can leave
+`);
+
+// Arrange
+plane3 = new Plane(`plane3`);
+airport.landPlane(plane3);
+airport.landPlane(plane);
+expected = 2;
+
+// Act
+airport.orderTakeOff(plane2);
+actual = airport.planesStanding.length;
+
+// Assert
+result = assertEquals(expected, actual);
+
+// Report
+console.log(`Test 9: Planes not at the airport can not leave: ${result}`);
+console.log(`Test 9: ${result ? `PASS` : `FAIL`}`);
+console.log(``);
+
+// Clean Up
+airport.planesStanding = [];
+result = undefined;
+actual = undefined;
+
```

Comment:
I am really happy with how much TDD I did, and I hope to continue in my next projects. I find it quite hard sometimes to put the logic down in code, so this is great practice.